### PR TITLE
feat(menu): support icons

### LINKED ISSUE #1388: md-menu with icons and colors
#### Bug, feature request, or proposal:

feature request
#### What is the expected behavior?

Correctly display icons inside md-menu.
Be able to add style using md-primary md-accent ... 
#### What is the current behavior?

icons are displayed too close of the text, maybe I don't use it correctly but if that's it then the docs needs to be updated.
It would also be great to change the style by using md-accent, md-primary ... classes as for md-button
#### What are the steps to reproduce?

```
<md-menu x-position="before" #globalMenu="mdMenu">
            <button md-menu-item>
                <md-icon fontIcon="search"></md-icon>
                globalMenu.apps
            </button>
            <button md-menu-item class="md-primary">
                <md-icon fontIcon="search"></md-icon>
                globalMenu.account
            </button>
</md-menu>
```

Providing a Plunker (or similar) is the _best_ way to get the team to see your issue.
Plunker template: http://plnkr.co/edit/o077B6uEiiIgkC0S06dd
I didn't manage to make md-menu works with the plunker sorry
#### Which versions of Angular, Material, OS, browsers are affected?

Angular 2.0.1
Material 2.0.0-alpha.9-2

### PR DESCRIPTION
**Pull request by kara**
This PR adds CSS support for icons.

Closes #1388 

r: @hansl

cc: @jelbourn

### CODE REVIEW

File: src/lib/menu/README.md [hunk 1]
```diff
@@ -84,6 +84,34 @@ class MyComp {
...
 
+### Adding an icon
+
+Menus also support displaying `md-icon` elements before the menu item text.
+
+*my-comp.html*
+```html
+<md-menu #menu="mdMenu">
+    <button md-menu-item> 
+        <md-icon> dialpad </md-icon>
```
Comment (jelbourn) [inline src/lib/menu/README.md]: Why so much indent?

File: src/lib/menu/_menu-theme.scss [hunk 2]
```diff
@@ -18,6 +18,10 @@
       color: md-color($foreground, 'disabled');
     }
 
+    md-icon {
+      color: md-color($foreground, 'icon');
```
Comment (jelbourn) [inline src/lib/menu/_menu-theme.scss]: Why is this necessary?

File: src/lib/core/style/_menu-common.scss [hunk 3]
```diff
@@ -42,4 +42,12 @@ $md-menu-vertical-padding: 8px !default;
   &[disabled] {
     cursor: default;
   }
+
+  md-icon {
```
Comment (jelbourn) [inline src/lib/core/style/_menu-common.scss]: Is there a link to the spec we could include here for this 16px value?

File: src/lib/menu/README.md (hunk 1 above)
Comment (kara) [inline src/lib/menu/README.md]: Inattention :) Will fix

File: src/lib/menu/_menu-theme.scss (hunk 2 above)
Comment (kara) [inline src/lib/menu/_menu-theme.scss]: It's the color in the spec and in Material 1's version. It looks pretty dramatic without it.

File: src/lib/core/style/_menu-common.scss (hunk 3 above)
Comment (kara) [inline src/lib/core/style/_menu-common.scss]: No, as far as I could tell, no exact measurement for this case in the spec. This was the value in Material 1.

**Comment by jelbourn** [comment]
LGTM
